feat: add new parent's option to clone

HunteRoi opened the pull request:
Decide whether to clone the parent or not as a base for the child auto-created channel. Defaults to false.

Fixes #10

Commit:
feat: add new parent's option to clone

Decide whether to clone the parent or not as a base for the child
auto-created channel. Defaults to false.

HunteRoi commented:
Hey! What's your take on this? @Ballaual

Ballaual commented:
@HunteRoi 
will check this out later this evening and give you feedback after testing

edit 14:32
tests working fine and as intended.
false = using category permissions
true = using permissions of the channel you joined in

But personally i'd change the logic behind the true parameter to fully clone the parent channel except the name.
Docs might need an update too if you'd like to implement it like this.
See https://github.com/HunteRoi/discord-temp-channels/pull/12

![false](https://user-images.githubusercontent.com/38478976/231763098-f1f099a5-8d2e-4a0d-bd08-a437bb8abd85.jpg)
![true](https://user-images.githubusercontent.com/38478976/231763106-c3ef89cb-171a-420c-9f3f-2028ea16a178.jpg)
![options](https://user-images.githubusercontent.com/38478976/231763109-ed5c9bbd-d74d-4b64-bc40-ddf2e985219e.jpg)

Ballaual committed:
edit: remove option parameters when using childShouldBeACopyOfParent: true (#12)

This would result that the channel will be a full clone of the parent channel exept the naming pattern. In my opinion  this might give that new added parameter more use of it.

Commit:
docs: update API doc for new parameter

Commit:
chore: add build workflow
